invisible_cities/io/mcinfo_io.py

```diff
@@ -173,20 +177,180 @@ def load_mchits(file_name: str,
     return mchits_dict
 
 
+def load_mchits_fromstr(file_name : str) -> pd.DataFrame:
+    """
+    Opens file and calls load_mchits_df
+    file_name : str
+                The name of the file to be read
+    """
+    extents = pd.read_hdf(file_name, 'MC/extents')
+    with tb.open_file(file_name) as h5in:
+        hits = load_mchits_df(h5in, extents)
+
+    return hits
+
+
+def load_mchits_df(h5in    : tb.file.File,
+                   extents : pd.DataFrame) -> pd.DataFrame:
+    """
+    Loads the MC hit information into a pandas DataFrame.
+    h5in    : pytables file
+              A file already read into a pytables object
+    extents : pd.DataFrame
+              The extents table from the file which gives
+              information about the event tracture.
+    """
+
+    hits_tb  = h5in.root.MC.hits
+
+    # Generating hits DataFrame
+    hits = pd.DataFrame({'hit_id'      : hits_tb.col('hit_indx'),
+                         'particle_id' : hits_tb.col('particle_indx'),
+                         'label'       : hits_tb.col('label').astype('U13'),
+                         'time'        : hits_tb.col('hit_time'),
+                         'x'           : hits_tb.col('hit_position')[:, 0],
+                         'y'           : hits_tb.col('hit_position')[:, 1],
+                         'z'           : hits_tb.col('hit_position')[:, 2],
+                         'energy'      : hits_tb.col('hit_energy')})
+
+    evt_hit_df = extents[['last_hit', 'evt_number']]
+    evt_hit_df.set_index('last_hit', inplace = True)
+
+    hits = hits.merge(evt_hit_df          ,
+                      left_index  =   True,
+                      right_index =   True,
+                      how         = 'left')
+    hits.rename(columns={"evt_number": "event_id"}, inplace = True)
+    hits.event_id.fillna(method='bfill', inplace = True)
+    hits.event_id = hits.event_id.astype(int)
+
+    # Setting the indexes
+    hits.set_index(['event_id', 'particle_id', 'hit_id'], inplace=True)
+
+    return hits
+
+
 def load_mcparticles(file_name: str,
                      event_range=(0, int(1e9))) -> Mapping[int, Mapping[int, MCParticle]]:
 
     with tb.open_file(file_name, mode='r') as h5in:
         return read_mcinfo(h5in, event_range)
 
 
+def load_mcparticles_df(iFileName: str) -> pd.DataFrame:
```

Comment:
Why don't you use two different functions as you do with `hits`? One that takes as an argument the file name and the other one that takes as an argument the result of opening the table. I would expect that if they're useful for one table, they could be for the other one, too.

Reply:
I added a function of this type for load_mcparticles_df but not for the sensors version.

Since the sensor info is read directly using pandas.read_hdf a tables.File object is never created so it seemed redundant to have another version